src/main/java/me/jellysquid/mods/sodium/client/render/chunk/compile/buffers/ChunkModelBuffers.java

```diff
@@ -1,8 +1,11 @@
 package me.jellysquid.mods.sodium.client.render.chunk.compile.buffers;
 
 import me.jellysquid.mods.sodium.client.model.quad.properties.ModelQuadFacing;
+import me.jellysquid.mods.sodium.client.render.chunk.data.ChunkRenderData;
 import me.jellysquid.mods.sodium.client.render.chunk.format.ModelVertexSink;
 
 public interface ChunkModelBuffers {
     ModelVertexSink getSink(ModelQuadFacing facing);
+
+    ChunkRenderData.Builder getRenderData();
```

Comment:
This could be pass into `ChunkModelVertexTransformer` instead. You can expand `ModelVertexSink#writeQuad` (I'm sure there is a better approach than this) to get `Sprite` and perform a check there.